vault/plugin_catalog_test.go

```diff
@@ -180,3 +183,98 @@ func TestPluginCatalog_List(t *testing.T) {
 		}
 	}
 }
+
+func TestPluginCatalog_NewPluginClient(t *testing.T) {
+	core, _, _ := TestCoreUnsealed(t)
+
+	sym, err := filepath.EvalSymlinks(os.TempDir())
+	if err != nil {
+		t.Fatalf("error: %v", err)
+	}
+	core.pluginCatalog.directory = sym
+
+	if len(core.pluginCatalog.externalPlugins) != 0 {
+		t.Fatalf("expected externalPlugins map to be of len 0 but got %d", len(core.pluginCatalog.externalPlugins))
+	}
```

Comment:
Totally a nit, but I like this little trick to shorten things a bit:
```suggestion
	if extPlugins := len(core.pluginCatalog.externalPlugins); extPlugins != 0 {
		t.Fatalf("expected externalPlugins map to be of len 0 but got %d", extPlugins)
	}
```
You can do something similar with the other length-based checks in this test if you'd like.